src/FrameHelper.cpp

```diff
@@ -685,6 +707,45 @@ namespace frame_helper
             }
             break;
         } // end case MODE_JPEG
+
+        case MODE_PNG:
+        {
+            //PNG --> RGB
+            switch(dst.getFrameMode())
+            {
+            case MODE_RGB:
+                {
+                    cv::Mat out = cv::imdecode(src.image,CV_LOAD_IMAGE_COLOR);
+                    dst.init(out.cols,out.rows,8,MODE_RGB);
+                    cv::cvtColor(out,FrameHelper::convertToCvMat(dst),cv::COLOR_BGR2RGB);
+                    break;
+                }
+
+                //PNG --> BGR
+            case MODE_BGR:
+                {
+                    cv::Mat out = cv::imdecode(src.image,CV_LOAD_IMAGE_COLOR);
+                    copyMatToFrame(out,dst);
+                    break;
+                }
+
+                // PNG --> PNG
+            case MODE_PNG:
+                dst.init(src,true);
+                break;
+
+                //PNG--> grayscale
+            case MODE_GRAYSCALE:
+                {
+                    cv::Mat out = cv::imdecode(src.image,0);
+                    copyMatToFrame(out,dst);
+                    break;
+                }
+
+            default:
+                throw std::runtime_error("FrameHelper::convertColor: Cannot convert frame mode png - mode is unknown");
```

Comment:
the message looks confusing to me ... it seems that PNG is the mode that is unknown